service/storage_service.cc

```diff
@@ -1936,17 +1934,20 @@ future<> storage_service::decommission() {
             std::unordered_set<gms::inet_address> nodes_down;
             auto req = node_ops_cmd_request{node_ops_cmd::decommission_prepare, uuid, ignore_nodes, leaving_nodes, {}};
             try {
-                parallel_for_each(nodes, [&ss, &req, &nodes_unknown_verb, &nodes_down, uuid] (const gms::inet_address& node) {
-                    return ss._messaging.local().send_node_ops_cmd(netw::msg_addr(node), req).then([uuid, node] (node_ops_cmd_response resp) {
+                ss.send_node_ops_cmd_wrapped(nodes, req, [uuid, &nodes_unknown_verb, &nodes_down] (const gms::inet_address& node,future<node_ops_cmd_response> resp_fut) {
```

Comment:
To consider: there's potential to deduplicate the code even further. All 4 handlers do pretty much the same thing, except that they log a slightly different message. So instead of `send_node_ops_cmd_wrapped` you can actually have a helper that takes the following parameters: the verb ("decommission", "bootstrap", "replace", "removenode"), the list of unknown verbs and the list of nodes down, and works like that:
```cpp
try {
    resp_fut.get();
    slogger.debug("{}[{}]: Got prepare response from node={}", verb, uuid, node);
} catch (seastar::rpc::unknown_verb_error&) {
    slogger.warn("{}[{}]: Node {} does not support {} verb", verb, uuid, node, verb);
    nodes_unknown_verb.emplace(node);
} catch (seastar::rpc::closed_error&) {
    slogger.warn("{}[{}]: Node {} is down for node_ops_cmd verb", verb, uuid, node);
    nodes_down.emplace(node);
}
return make_ready_future<>();
```

Reply:
I will do this change if I keep the code.